star_distribution.py

```diff
@@ -0,0 +1,119 @@
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+
+# define the rotation curve
+def vel(r1):
+
+    ar1 = math.fabs(r1)
+    if (ar1 <= 5.0):
+        v1 = 1.4*ar1*math.exp(-ar1/20.0)
+    elif (ar1 > 5.0 and ar1 < 6.0):
+        max = 1.4*5*math.exp(-5.0/20.0)
+        min = 5.0
+        m = (max - min)/(-1.0)
+        v1 = m*(ar1 - 5.0) + max
+    else:
+        v1 = 5.0
+
+    return v1
+
+def vel(r2):
+
+    ar2 = math.fabs(r2)
+    if (ar2 <= 5.0):
+        v2 = 1.4*ar2*math.exp(-ar2/20.0)
+    elif (ar2 > 5.0 and ar2 < 6.0):
+        max = 1.4*5*math.exp(-5.0/20.0)
+        min = 5.0
+        m = (max - min)/(-1.0)
+        v2 = m*(ar2 - 5.0) + max
+    else:
+        v2 = 5.0
+
+    return v2
+
+def vel(r3):
+
+    ar3 = math.fabs(r3)
+    if (ar3 <= 5.0):
+        v3 = 1.4*ar3*math.exp(-ar3/20.0)
+    elif (ar3 > 5.0 and ar3 < 6.0):
+        max = 1.4*5*math.exp(-5.0/20.0)
+        min = 5.0
+        m = (max - min)/(-1.0)
+        v3 = m*(ar3 - 5.0) + max
+    else:
+        v3 = 5.0
+
+    return v3
+
+
+
+def spiral():
+
+    rmin = -20
+    rmax = 20
+    npts1 = 16
+    npts2 = 8
+    npts3 = 4
+
+    r1 = np.arange(npts1, dtype=np.float64)*(rmax - rmin)/(npts1 - 1.0) + rmin
+    r2 = np.arange(npts2, dtype=np.float64) * (rmax - rmin) / (npts2 - 1.0) + rmin
+    r3 = np.arange(npts3, dtype=np.float64) * (rmax - rmin) / (npts3 - 1.0) + rmin
+
+    omega = np.zeros( (npts1) )
+    gamma = np.zeros( (npts2) )
+    beta = np.zeros( (npts3) )
+
+    n1 = 0
+    while (n1 < npts1):
+        omega[n1] = math.fabs(vel(r1[n1])/r1[n1])
```

Comment:
python is special in how you can handle lists - especially if you make the list a numpy array. 

As you defined omega=np.zeros(npts1), it already is a numpy array. If you make v1 and r1 also np arrays with v1=np.array(v1), then you can forgo this loop entirely and the definition of omega and initialize omega simply with omega=np.abs(vel(r1)/r1) .

Learn this and use it to avoid loops whenever possible.